src/components/Modal/AddMailModal.tsx

```diff
@@ -45,31 +52,24 @@ export const AddMailModal = withRouter((props: IAddMailModalProps) => {
   const alert = useAlert();
   const logoutModalContext = useContext(LogoutModalContext);
 
-  const [newMail, setNewMail] = useState("");
-  const [confirmMail, setConfirmMail] = useState("");
+  const { register, handleSubmit, getValues, errors } = useForm<
+    IChangeMailFormData
+  >();
 
   /*
    * Boolean variable to check if a work email different from spid one is already set
    * */
-  const isWorkMailNotSet = !props.workMail || props.workMail === props.spidMail;
+  const isWorkMailNotSet =
+    (!props.workMail || props.workMail === props.spidMail) &&
+    props.location.pathname !== "/profile";
```

Comment:
why we need the condition `props.location.pathname !== "/profile";` to check if the work email is not set ?

Reply:
it was actually wrong, it is an additional condition to check to change modal/button texts based on location from where I open the modal; it is now fixed